Tests/NIOCoreTests/ByteBufferTest.swift

```diff
@@ -1904,6 +1904,33 @@ class ByteBufferTest: XCTestCase {
         let expected = "00 01 02 03 04 ... fb fc fd fe ff"
         XCTAssertEqual(expected, actual)
     }
+    
+    func testHexDumpCompact() {
+        let buf = ByteBuffer(string: "Hello")
+        XCTAssertEqual("48656c6c6f", buf.hexDump(format: .compact))
+    }
+
+    func testHexDumpCompactEmptyBuffer() {
+        let buf = ByteBuffer(string: "")
+        XCTAssertEqual("", buf.hexDump(format: .compact))
+    }
+
+    func testHexDumpCompactWithReaderIndexOffset() {
+        var buf = ByteBuffer(string: "Hello")
+        let firstTwo = buf.readBytes(length: 2)!
+        XCTAssertEqual([72, 101], firstTwo)
+        XCTAssertEqual("6c6c6f", buf.hexDump(format: .compact))
+    }
+
+    func testHexDumpCompactWithMaxBytes() {
+        self.buf.clear()
+        for f in UInt8.min...UInt8.max {
+            self.buf.writeInteger(f)
+        }
+        let actual = self.buf.hexDump(format: .compact(maxBytes: 10))
+        let expected = "0001020304 ... fbfcfdfeff"
```

Comment:
This is a completely subjective, style question here for the output but... this variant is _compact_ and documents itself as being "like plain, but without the whitespace". With that in mind, how do we feel about the whitespace around the `...`, when `readableBytes > maxBytes`? Should we have: 

1. `0001020304 ... fbfcfdfeff`; or
2. `0001020304...fbfcfdfeff`?

I'm just parking this here for discussion, so no need to action this right away @supersonicbyte, but given it's API we might like to make a deliberate decision on it.

//cc @weissi @FranzBusch

Reply:
Anyways, just let me know for what option we will go so I can fix it accordingly. Thanks!

Reply:
I think no whitespace would be great indeed. The initial reason I asked for this was for a command line too that outputs:

```
key-hex-bytes value-hex-bytes
```

and it was important that the `...-hex-bytes` could be read with

```
while read -r key value; do
   ...
done
```

so having whitespace in either key or value would break that code. Similarly, if you would use `cut -d' ' -f1` or so you wouldn't want any whitespace there.

We should probably also add to the docs that this format guarantees to never emit whitespace.

Thank you!!

Reply:
Alright, as per @weissi answer I went ahead and implemented the option 2. Feel free to review and give feedback. Thanks!

Reply:
Will also be slightly nicer for all the `...`s to instead be `…`. The former is 3 unicodes (3 `.`s), the latter is the 1 unicode character, the "U+2026 HORIZONTAL ELLIPSIS".